atc/exec/task_step.go

```diff
@@ -222,16 +222,18 @@ func (step *TaskStep) Run(ctx context.Context, state RunState) error {
 		step.lockFactory,
 	)
 
-	err = result.Err
 	if err != nil {
-		if err == context.Canceled || err == context.DeadlineExceeded {
+		switch err {
+		case context.Canceled:
+		case context.DeadlineExceeded:
```

Comment:
if I understood correctly here, you wanted to get the outputs registered in both `context.Canceled` and `context.DeadlineExceed`, right?

if that's the case, the `case`ing here is wrong I think 🤔  

```go
package main

func run(arg string) (result string) {
        result = "nothing"

        switch arg {
        case "foo":
        case "bar":
                result = "xxxxx"
        case "caz":
                result = "-----"
        }

        return
}

func main() {
        println(run("foo"))
        println(run("bar"))
        println(run("caz"))
}
```



```
nothing
xxxxx
-----
```

if that's indeed a regression, I wonder if we're missing a test for the output registration

Reply:
thats definitely what we were going for, I was curious why a test didn't fail.

Reply:
https://github.com/concourse/concourse/pull/4987/files#diff-a1aa3a540c36b7ed6dc5f979aea7b6e1R920-R935

Reply:
add coverage to ensure we test output registration in the event of ctx.deadline or ctx.cancel errors

Reply:
done